pkg/services/sqlstore/migrations/accesscontrol/datasource_drilldown_removal.go

```diff
@@ -0,0 +1,118 @@
+package accesscontrol
+
+import (
+	"fmt"
+	"strings"
+
+	"xorm.io/xorm"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+const (
+	datasourceDrilldownRemoval = "delete all drilldown actions, their assignments, and related roles"
+)
+
+func AddDatasourceDrilldownRemovalMigration(mg *migrator.Migrator) {
+	mg.AddMigration(datasourceDrilldownRemoval, &datasourceDrilldownRemovalMigrator{})
+}
+
+type datasourceDrilldownRemovalMigrator struct {
+	migrator.MigrationBase
+}
+
+func (m *datasourceDrilldownRemovalMigrator) SQL(dialect migrator.Dialect) string {
+	return CodeMigrationSQL
+}
+
+func (m *datasourceDrilldownRemovalMigrator) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
+	var roleIDs []int64
+	err := sess.SQL("SELECT DISTINCT role_id FROM permission WHERE action LIKE ?", "%drilldown%").Find(&roleIDs)
```

Comment:
Just wondering, is `%drilldown%` good enough or can we get more close to just `datasources:drilldown`?

Reply:
we could be more granular to have datasources. we have both fixed_ plugins and managed permissions. il change to have only datasources:drilldown 👍

Reply:
I'm just afraid this migration would also catch any action that's being added in the future 😅 Thanks for your work here, Eric!